src/base/bittorrent/session.cpp

```diff
@@ -1931,11 +1948,20 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
     if (!fromMagnetUri) {
         if (params.restored) {
             // Set torrent fast resume data
-            p.resume_data = {fastresumeData.constData(), fastresumeData.constData() + fastresumeData.size()};
+            p.resume_data = std::vector<char> {fastresumeData.constData(), fastresumeData.constData() + fastresumeData.size()};
             p.flags |= libt::add_torrent_params::flag_use_resume_save_path;
         }
         else {
-            p.file_priorities = {params.filePriorities.begin(), params.filePriorities.end()};
+            std::transform(params.filePriorities.cbegin(), params.filePriorities.cend()
```

Comment:
just want to make sure that `p.file_priorities` will be guaranteed to be empty before here right?

Reply:
It is default constructed std::vector so it is empty.

Reply:
Or you can add `p.file_priorities.clear()` to guarantee it will be empty?

Reply:
Isn't that too much? What specifically bothers you that you think it might not be empty?
>since there are a lot of code between this line and p declaration.

It doesn't touch `file_priorities` field (at least it shouldn't touch this field). Anyway clearing it here is a way to hide a possible logical error.
If you're really concerned about it, I can add `Q_ASSERT(p.file_priorities.empty())`.

Reply:
>If you're really concerned about it, I can add Q_ASSERT(p.file_priorities.empty()).

That would be great.

Reply:
Ok. Done.
PR updated.